third_party/terraform/utils/iam_iap_tunnel.go.erb

```diff
@@ -0,0 +1,188 @@
+<% autogen_exception -%>
+package google
+<% unless version == 'ga' -%>
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/cloudresourcemanager/v1"
+	iap "google.golang.org/api/iap/v1beta1"
+)
+
+var IamIapTunnelSchema = map[string]*schema.Schema{
+	"instance": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+
+	"project": {
+		Type:     schema.TypeString,
+		Optional: true,
+		Computed: true,
+		ForceNew: true,
+	},
+
+	"zone": {
+		Type:     schema.TypeString,
+		Optional: true,
+		Computed: true,
+		ForceNew: true,
+	},
+}
+
+type IapTunnelIamUpdater struct {
+	project       string
+	zone          string
+	resourceId    string
+	projectNumber string
+	Config        *Config
+}
+
+func NewIapTunnelIamUpdater(d *schema.ResourceData, config *Config) (ResourceIamUpdater, error) {
+	project, err := getProject(d, config)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := config.clientResourceManager.Projects.Get(project).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	projectNumber := strconv.FormatInt(res.ProjectNumber, 10)
```

Comment:
From the documentation it looked like it relied on project number, so that is what I followed it. However, you make a good point, I didn't actually test if project_id would work instead. I'll do a quick test of that and see if it works (and if it does I'll switch over the code to use that instead)

Reply:
Thanks! That's curious that they decided to use the project number there in the docs when the id has historically been preferred in URIs; my suspicion is that both work, since most other resources that expect id can accept a # too.

Reply:
@rileykarson just did a test and swapped out number for id, and get a 400 back from the backend saying that its an invalid argument. So from my testing i think we're stuck with projectNumber. This is just an implementation detail though that is hidden from the user, so I think we'll be safe to stick with projectnumber for now. But if the API updates to projectID, should be an easy swap.

Reply:
Sounds good! Thanks for trying that out. Do you mind commenting with that context here, that the API only accepts project numbers and sends a 400 back if we send an id?